File: .tmp_py4013/batch_11_dump/3256_maximum_value_sum_by_placing_three_rooks_i.js
// LeetCode 3256 - Maximum Value Sum by Placing Three Rooks I
// https://leetcode.com/problems/maximum-value-sum-by-placing-three-rooks-i/

var maximumValueSum = function(board) {
    const m = board.length, n = board[0].length;
    const tops = [];
    for (let i = 0; i < m; i++) {
        const row = [];
        for (let j = 0; j < n; j++) {
            const cur = {v: board[i][j], c: j};
            let placed = false;
            for (let t = 0; t < row.length; t++) {
                if (cur.v > row[t].v) { row.splice(t, 0, cur); placed = true; break; }
            }
            if (!placed) row.push(cur);
            if (row.length > 3) row.length = 3;
        }
        tops.push(row);
    }
    let ans = Number.MIN_SAFE_INTEGER;
    for (let i = 0; i < m; i++) {
        for (const a of tops[i]) {
            for (let j = i + 1; j < m; j++) {
                for (const b of tops[j]) {
                    if (a.c === b.c) continue;
                    for (let k = j + 1; k < m; k++) {
                        for (const c of tops[k]) {
                            if (c.c === a.c || c.c === b.c) continue;
                            const s = a.v + b.v + c.v;
                            if (s > ans) ans = s;
                        }
                    }
                }
            }
        }
    }
    return ans;
};

File: .tmp_rs4013/rust_check15/3256_maximum_value_sum_by_placing_three_rooks_i.rs
struct Solution;
// LeetCode 3256 - Maximum Value Sum by Placing Three Rooks I
// https://leetcode.com/problems/maximum-value-sum-by-placing-three-rooks-i/

impl Solution {
    pub fn maximum_value_sum(board: Vec<Vec<i32>>) -> i64 {
        let m = board.len();
        let n = board[0].len();
        let mut tops: Vec<Vec<(i32, usize)>> = Vec::new();
        for i in 0..m {
            let mut row: Vec<(i32, usize)> = Vec::new();
            for j in 0..n {
                let cur = (board[i][j], j);
                let mut placed = false;
                for t in 0..row.len() {
                    if cur.0 > row[t].0 {
                        row.insert(t, cur);
                        placed = true;
                        break;
                    }
                }
                if !placed {
                    row.push(cur);
                }
                if row.len() > 3 {
                    row.truncate(3);
                }
            }
            tops.push(row);
        }
        let mut ans = i64::MIN / 2;
        for i in 0..m {
            for &(av, ac) in &tops[i] {
                for j in i + 1..m {
                    for &(bv, bc) in &tops[j] {
                        if ac == bc {
                            continue;
                        }
                        for k in j + 1..m {
                            for &(cv, cc) in &tops[k] {
                                if cc == ac || cc == bc {
                                    continue;
                                }
                                let s = av as i64 + bv as i64 + cv as i64;
                                if s > ans {
                                    ans = s;
                                }
                            }
                        }
                    }
                }
            }
        }
        ans
    }
}

fn main() {}
